// Copyright 2016 Rodrigo Jose Hernandez Cordoba
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package AeonGames;

message Vector3Buffer
{
    float x = 1;
    float y = 2;
    float z = 3;
}

message MeshBuffer {
	uint32 Version         = 1;
    Vector3Buffer Min      = 2;
    Vector3Buffer Max      = 3;
    uint32 VertexFlags     = 4;
    uint32 VertexCount     = 5;
    uint32 IndexType       = 6;		
    uint32 IndexCount      = 7;
    bytes VertexBuffer     = 8;
}
